msal/src/main/java/com/microsoft/identity/client/Account.java

```diff
@@ -22,73 +22,102 @@
 //  THE SOFTWARE.
 package com.microsoft.identity.client;
 
-/**
- * MSAL API surface implementation of an account.
- */
-class Account implements IAccount {
-
-    private IAccountIdentifier mAccountIdentifier;
-    private IAccountIdentifier mHomeAccountIdentifier;
-    private String mUsername;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import com.microsoft.identity.common.internal.logging.Logger;
+import com.microsoft.identity.common.internal.providers.microsoft.MicrosoftIdToken;
+import com.nimbusds.jwt.JWT;
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.JWTParser;
+
+import java.text.ParseException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class Account implements IAccount {
+
+    private static final String TAG = Account.class.getSimpleName();
+
+    private final String mRawIdToken;
+    private String mHomeOid;
+    private String mHomeTenantId;
     private String mEnvironment;
 
-    Account() {
-        // Empty constructor
+    public Account(@Nullable final String rawIdToken) {
+        mRawIdToken = rawIdToken;
     }
 
-    /**
-     * Sets the account id.
-     *
-     * @param accountId The IAccountIdentifier to set.
-     */
-    void setAccountIdentifier(final IAccountIdentifier accountId) {
-        mAccountIdentifier = accountId;
+    void setId(@Nullable final String id) {
+        mHomeOid = id;
     }
 
+    @NonNull
     @Override
-    public IAccountIdentifier getAccountIdentifier() {
-        return mAccountIdentifier;
-    }
+    public String getId() {
+        String id;
+
+        if (null != mRawIdToken && null != getClaims()) {
+            id = (String) getClaims().get(MicrosoftIdToken.OBJECT_ID);
+        } else {
+            id = mHomeOid;
```

Comment:
Why are we returning Home OID when OID is not there? Trying to understand the scenario here.

Reply:
This is supposed to be the home tenant object id if available.  Let's review the identifier situation together.

Reply:
When there is no idtoken present (say, in the case where a guest has authorized without authorizing to their home tenant) I am populating this value with the oid returned from `client_info`

Reply:
I think that Account Id should always be home object id though.

Reply:
Yep, it [currently] is